### DISCUSSION TITLE
Anyone who can help optimize my code for generating dynamic diagrams?

### DISCUSSION THREAD

**Discussion by minitaylor** [clientIO/joint#1886] [Q&A]
I need help to improve the "look" of the connections and to have someone expert clean up my code, offer me other alternative ways of doing things. Despite my efforts, the movement of the connections is not ideal.
The first time the hierarchical diagram with its connections is drawn, it usually has a few "errors" in the sense that:
- Some Lines may go behind boxes
- Some lines may connect to the wrong side of the box (Connections are always from lower boxes to higher boxes; therefore arrows should "leave" from the top side and "arrive" at the bottom side
Also my SVG export is not exactly the same as the map; a few objects are lost.
![ScreenShot 2022-11-14 at 11 48 30 AM](https://user-images.githubusercontent.com/13508031/201628924-c67eb934-3eff-405b-b11e-e39e2c39cdb6.png)

**Comment by marekhozak10** [comment]
Hi @minitaylor,

Thank you for reaching out. We would need to understand your context a little better so we can point you to the right resources. Do you already have a JointJS+ commercial license?

Our priority support subscriptions are here to help you kickstart your project with maximum priority, guaranteed response time, code reviews from the very same engineers who developed JointJS+ and with whom you will be able to book conference calls as needed.

Should you be interested in our support subscriptions, please write an email to alexis@jointjs.com who will be more than happy to help you.

Cheers!

Marek from JointJS
